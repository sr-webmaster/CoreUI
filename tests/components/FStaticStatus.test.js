import { mount } from '@vue/test-utils'
import Component from '@freshinup/core-ui/src/components/FStaticStatus.vue'

describe('FStaticStatus', () => {
  it('snapshot default', async () => {
    const wrapper = mount(Component)
    expect(wrapper.element).toMatchSnapshot()
  })
  it('snapshot with value', async () => {
    const wrapper = mount(Component, {
      propsData: {
        value: 'inactive'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('snapshot with custom statuses', async () => {
    const wrapper = mount(Component, {
      propsData: {
        value: 1,
        statuses: [
          { id: 1, name: 'Draft', color: 'accent' },
          { id: 2, name: 'Pending', color: 'warning' },
          { id: 3, name: 'Confirmed', color: 'success' },
          { id: 4, name: 'Past', color: 'secondary' },
          { id: 5, name: 'Cancelled', color: 'accent' }
        ]
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
  it('null activeItem on unmatched value', async () => {
    const wrapper = mount(Component, {
      propsData: {
        value: 'willNotBeMatched'
      }
    })
    expect(wrapper.vm).toHaveProperty('activeItem', null)
  })
  it('computed label', async () => {
    const wrapper = mount(Component, {
      propsData: {
        value: 'inactive'
      }
    })
    expect(wrapper.vm).toHaveProperty('label', 'Inactive')
  })
  it('computed label by name from passed in statuses', async () => {
    const wrapper = mount(Component, {
      propsData: {
        value: 3,
        statuses: [
          { id: 1, name: 'Draft', color: 'accent' },
          { id: 2, name: 'Pending', color: 'warning' },
          { id: 3, name: 'Confirmed', color: 'success' },
          { id: 4, name: 'Past', color: 'secondary' },
          { id: 5, name: 'Cancelled', color: 'accent' }
        ]
      }
    })
    expect(wrapper.vm).toHaveProperty('label', 'Confirmed')
  })
})
