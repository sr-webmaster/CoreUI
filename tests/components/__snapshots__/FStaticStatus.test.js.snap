// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FStaticStatus snapshot default 1`] = `
<span
  class="v-btn v-btn--disabled success successtext--text"
>
  Active
</span>
`;

exports[`FStaticStatus snapshot with custom statuses 1`] = `
<span
  class="v-btn v-btn--disabled accent accenttext--text"
>
  Draft
</span>
`;

exports[`FStaticStatus snapshot with value 1`] = `
<span
  class="v-btn v-btn--disabled error errortext--text"
>
  Inactive
</span>
`;
